src/hotspot/cpu/zero/frame_zero.cpp

```diff
@@ -116,13 +116,67 @@ void frame::patch_pc(Thread* thread, address pc) {
 }
 
 bool frame::safe_for_sender(JavaThread *thread) {
-  ShouldNotCallThis();
-  return false;
+  address sp = (address)_sp;
+
+  // consider stack guards when trying to determine "safe" stack pointers
+  // sp must be within the usable part of the stack (not in guards)
+  if (!thread->is_in_usable_stack(sp)) {
+    return false;
+  }
+
+  // an fp must be within the stack and above (but not equal) sp
+  if (!thread->is_in_stack_range_excl((address)fp(), sp)) {
+    return false;
+  }
+
+  // All good.
+  return true;
 }
 
 bool frame::is_interpreted_frame_valid(JavaThread *thread) const {
-  ShouldNotCallThis();
-  return false;
+  assert(is_interpreted_frame(), "Not an interpreted frame");
+  // These are reasonable sanity checks
+  if (fp() == 0 || (intptr_t(fp()) & (wordSize-1)) != 0) {
```

Comment:
Use `is_aligned()` here?

Reply:
I could, but this matches what other platforms are doing in their `frame::is_interpreted_frame_valid()`. If there are no other fixes needed, okay if I keep this one in place? Otherwise, I would need to re-test the whole thing for a minor touchup, which is tedious.

Reply:
@theRealAph Do you agree with above? Any more comments?

Reply:
Paging @theRealAph ;)

Reply:
OK, but it seems weird.